src/Lecture1.hs

```diff
@@ -108,7 +128,20 @@ and finds a sum of the numbers inside this string.
 
 The string contains only spaces and/or numbers.
 -}
-strSum str = error "TODO"
+strSum :: String -> Int
+strSum str =
+    let l = words str
+    in sumStrList l
+
+sumStrList :: [String] -> Int
+sumStrList l =
+    if null l 
+    then 0
+    else    
+        let
+            s = read (head l) :: Int
```

Comment:
As you specify the top-level type signature for this function, the compiler would be able to figure out what type `read` function has in this case due to type inference, so you can skip specifying it in the body:

```suggestion
            s = read (head l)
```